src/crewai/agents/executor.py

```diff
@@ -185,6 +197,52 @@ def _iter_next_step(
             yield AgentStep(action=output, observation=observation)
             return
 
+        except BadRequestError as e:
+            print("Bad Request Error", e)
+            if "context_length_exceeded" in str(e):
+                self._logger.log(
+                    "debug",
+                    "Context length exceeded. Asking user if they want to use summarize prompt to fit, this will reduce context length.",
+                    color="yellow",
+                )
+                user_choice = prompt(
```

Comment:
_Suggestion_: In here again we could've been simpler

```suggestion
                user_choice = input("Context length exceeded. Do you want to summarize the text to fit models context window? (Y/N): ")
                validate_user_input(user_choice)
```